utilities/backupable/backupable_db.cc

```diff
@@ -1441,27 +1596,74 @@ Status BackupEngineImpl::VerifyBackup(BackupID backup_id,
     }
     if (verify_with_checksum) {
       // verify file checksum
-      std::string checksum_hex;
+      // try setting checksum_func
+      std::unique_ptr<FileChecksumGenerator> checksum_func;
+      std::string src_checksum_func_name = kUnknownFileChecksumFuncName;
+      std::string src_checksum_str = kUnknownFileChecksum;
+      std::string src_checksum_hex;
+      if (IsSstFile(file_info->filename)) {
+        // Extract table file number
+        const std::string& file = file_info->filename;
+        std::string local_name;
+        size_t last_slash = file.find_last_of('/');
+        assert(last_slash != std::string::npos);
+        local_name = file.substr(last_slash + 1);
+        if (file.substr(0, last_slash) == GetSharedChecksumDirRel()) {
+          local_name = GetFileFromChecksumFile(local_name);
+        }
+        uint64_t number;
+        FileType type;
+        bool ok = ParseFileName(local_name, &number, &type);
+        if (!ok) {
+          return Status::Corruption(
+              "Backup corrupted: Fail to parse filename " + local_name);
+        }
+        assert(type == kTableFile);
+        // Try to get checksum for the table file
+        Status s = checksum_list->SearchOneFileChecksum(
+            number, &src_checksum_str, &src_checksum_func_name);
+        if (s.ok() && src_checksum_str != kUnknownFileChecksum &&
+            src_checksum_func_name != kUnknownFileChecksumFuncName) {
+          // Ignore returned status of the following function.
+          // If checksum_func is nullptr, we verify with crc32c but not the
+          // custom checksum function.
+          SetChecksumFunc(src_checksum_func_name, checksum_func);
+          src_checksum_hex = ChecksumStrToHex(src_checksum_str);
+        }
+      }
+
       ROCKS_LOG_INFO(options_.info_log, "Verifying %s checksum...\n",
                      abs_path.c_str());
+      std::string checksum_hex;
+      std::string custom_checksum_hex;
       CalculateChecksum(abs_path, backup_env_, EnvOptions(), 0 /* size_limit */,
-                        &checksum_hex);
+                        &checksum_hex, checksum_func, &custom_checksum_hex);
       if (file_info->checksum_hex != checksum_hex) {
         std::string checksum_info(
             "Expected checksum is " + file_info->checksum_hex +
             " while computed checksum is " + checksum_hex);
-        return Status::Corruption("File corrupted: Checksum mismatch for " +
+        return Status::Corruption("File corrupted: crc32c mismatch for " +
                                   abs_path + ": " + checksum_info);
       }
+      if (checksum_func != nullptr && src_checksum_hex != custom_checksum_hex) {
+        std::string checksum_info("Expected checksum is " + src_checksum_hex +
```

Comment:
Also point out the custom checksum verification failed in the message.

Reply:
Actually, I think custom checksum failure information is in the message :) (`src_checksum_func_name` is the custom checksum function name.)

Reply:
I see, thanks.